sql/logic_test.go

```diff
@@ -299,40 +299,50 @@ SET DATABASE = test;
 					query.label = fields[3]
 				}
 			}
+
 			var buf bytes.Buffer
 			for s.Scan() {
 				line := s.Text()
 				if line == "----" {
+					if query.expectErr != "" {
+						t.Fatalf("%s: invalid ---- line break after a query expecting an error: %s", query.pos, query.expectErr)
+					}
+					break
+				}
+				fields := strings.Fields(s.Text())
+				if len(fields) == 0 {
```

Comment:
No need to split into fields. This can be `if strings.TrimSpace(s.Text()) == ""`.